packages/api/src/routes/internal.ts

```diff
@@ -211,6 +214,38 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * GET /internal/cx-data/patients
+ *
+ * Returns the cx patients for a given facility used for internal scripts
+ * @param req.query.facilityId - The facility ID.
+ */
+router.get(
+  "/cx-data/patients",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const facilityId = getFrom("query").orFail("facilityId", req);
+    const patients = await getPatients({ cxId, facilityId });
+
+    const assessments = await Promise.all(
+      patients.map(async patient => {
+        return await getCoverageAssessment({ cxId, patient });
+      })
+    );
```

Comment:
A cx might have a lot of patients, we should prob limit the amount of promises/parallel executions we trigger here - could use `executeAsynchronously` for that.

Reply:
`executeAsynchronously` return void -- so went with chunking instead. Done.

Reply:
Send msg in slack -- still confused on this one. `getCoverageAssessment` returns something I use, where as `executeAsynchronously` doesn't keep track of the return values when it calls `await executeSynchronously`

```
  const promises = new Array(amountOfPromises).fill(0).map(async (_, promiseIndex) => {
    // possible jitter before each run so that they don't start at the same time
    const jitter = Math.max(minJitterMillis, Math.random() * maxJitterMillis);
    await sleep(jitter);

    await executeSynchronously(
      itemsToProcess,
      fn,
      promiseIndex,
      amountOfPromises,
      indexControl,
      keepExecutingOnError,
      log
    );
  });
```

Reply:
Updated with `executeAsynchronously` everywhere previously chunking.